File: apps/02-personal-md-files/backend/python/src/services/llm/llm_interface.py
from abc import ABC, abstractmethod

class LLMInterface(ABC):
    """Interface for LLM providers"""
    
    @abstractmethod
    async def send_message(self, message: str) -> str:
        """Send a message to the LLM provider and get a response"""
        pass
    
    @abstractmethod
    def get_provider(self) -> str:
        """Get the name of the LLM provider"""
        pass
    
    @abstractmethod
    def get_model(self) -> str:
        """Get the model name used by the LLM provider"""
        pass 

File: apps/02-personal-md-files/backend/golang/src/services/llm/llm_interface.go
package llm

// LLMInterface defines the interface that all LLM providers must implement
type LLMInterface interface {
	// SendMessage sends a message to the LLM provider and returns the response
	SendMessage(message string) (string, error)
	
	// GetProvider returns the name of the LLM provider
	GetProvider() string
	
	// GetModel returns the model name used by the LLM provider
	GetModel() string
} 